clients/venice-client/src/main/java/com/linkedin/venice/fastclient/transport/GrpcTransportClient.java

```diff
@@ -0,0 +1,191 @@
+package com.linkedin.venice.fastclient.transport;
+
+import com.google.protobuf.ByteString;
+import com.linkedin.venice.client.store.transport.TransportClient;
+import com.linkedin.venice.client.store.transport.TransportClientResponse;
+import com.linkedin.venice.compression.CompressionStrategy;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.protocols.VeniceClientRequest;
+import com.linkedin.venice.protocols.VeniceReadServiceGrpc;
+import com.linkedin.venice.protocols.VeniceServerResponse;
+import com.linkedin.venice.request.RequestHelper;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class GrpcTransportClient extends InternalTransportClient {
+  private static final Logger LOGGER = LogManager.getLogger(GrpcTransportClient.class);
+  private final TransportClient r2TransportClient; // used for metadata requests
+
+  // field has public visibility for unit testing
+  public final Map<String, ManagedChannel> serverGrpcChannels;
+  final Map<String, String> veniceAddressToGrpcAddress;
+
+  public GrpcTransportClient(Map<String, String> nettyServerToGrpc, TransportClient r2TransportClient) {
+    serverGrpcChannels = new HashMap<>();
+    veniceAddressToGrpcAddress = new HashMap<>();
+    this.r2TransportClient = r2TransportClient;
+
+    for (Map.Entry<String, String> entry: nettyServerToGrpc.entrySet()) {
+      String nettyServer = entry.getKey().split(":")[0];
+      String grpcServer = nettyServer + ":" + entry.getValue();
+
+      veniceAddressToGrpcAddress.put(entry.getKey(), grpcServer);
+    }
+  }
+
+  public ManagedChannel getChannel(String requestPath) {
+    String portKey = requestPath.split("/")[2];
+
+    if (!veniceAddressToGrpcAddress.containsKey(portKey)) {
+      throw new VeniceException("No grpc server found for port: " + portKey);
+    }
+
+    return serverGrpcChannels.computeIfAbsent(
+        portKey,
+        k -> ManagedChannelBuilder.forTarget(veniceAddressToGrpcAddress.get(k)).usePlaintext().build());
```

Comment:
So we can't use SSL here?

Reply:
we will be able to use SSL to create secure connections. In order to enable TLS on a server we will have to provide the Certificate Chain File and the Private Key file. From what I have seen, there are no functions within Venice wrt to SSL utils that provides the path to these files. It should be straightforward otherwise. One solution is to add additional fields within `SSLConfig` which would have the path of the necessary files. https://grpc.io/docs/guides/auth/